### DISCUSSION TITLE
User-specific ratings

### DISCUSSION THREAD

**Discussion by tyrrminal** [mealie-recipes/mealie#889] [Feature Request]
**Before submitting this feature request I have**
- [X] Reviewed the documentation and verified that my feature request isn't already a feature.
- [X] Reviewed the existing feature requests and my feature has not been requested.
- [X] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
Any logged in user is able to change the rating on a recipe, and their rating is then displayed to all users. Even if the system is generally only used by tight family groups, this still means that any one family member can control the rating of any recipe, which seems to go against the system's multi-user nature.

**Suggest A Solution**
Each user's rating should be stored for each recipe rated. This allows the user to see/change their own rating at any time, and allows any user to see an aggregated average rating for the recipe. Additional features could then be introduced around this functionality, for instance, low-rating filtering, so that a user could search for only recipes that everyone (in their Group) had rated at least 3-stars.

**Additional Information**
- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by vegardhw** [comment]
I would like to see this feature also as it makes much more sense in a family multi-user scenario.

**Comment by tbleiker** [comment]
I'd like to see thif feature as well. I am thinking of sharing my recipes with friends. However, they most probably have another taste then we (as family) have... 😄

**Comment by MrSaiclops** [comment]
Came here to request the very same thing. Immediately when showing it to my spouse they asked about getting average scores between the two of us.

**Comment by Desecrate2337** [comment]
I would like this feature as well! I signed up here to suggest it. It would be nice if it showed the rating each user gave, and also the average ratings between all users for the recipe.

**Comment by threadstone83** [comment]
I can only agree. I would really like this feature too!

**Comment by michael-genson** [comment]
While still a draft, see https://github.com/mealie-recipes/mealie/pull/3345

**Comment by ChipSkylark37** [comment]
Has this been implemented? I'm the primary User currently and have some others starting to use Mealie and would go over with them how this works. Cheers.

**Reply by michael-genson** [reply]
Yup! Merged last April
